spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -157,8 +162,10 @@ public Map<String, String> properties() {
     propsBuilder.put("provider", "iceberg");
     String currentSnapshotId = icebergTable.currentSnapshot() != null ?
         String.valueOf(icebergTable.currentSnapshot().snapshotId()) : "none";
-    propsBuilder.put("current-snapshot-id", currentSnapshotId);
+    propsBuilder.put(CURRENT_SNAPSHOT_ID, currentSnapshotId);
     propsBuilder.put("location", icebergTable.location());
+    int formatVersion = ((HasTableOperations) icebergTable).operations().current().formatVersion();
```

Comment:
This is used for other implementations as well that may not implement `HasTableOperations`. Instead, this should check whether `icebergTable` is an instance of `BaseTable` and add the property only if it is. `BaseTable` exposes the table operations.

Reply:
Kyle was suggesting `BaseTable` in Slack as well, but I was thinking `HasTableOperations` is an interface that always exposes the table operation. My hunch was wrong. Glad to know & updated the code. Thank you.